holidays/countries/yemen.py

```diff
@@ -0,0 +1,206 @@
+#
+#  Authors: Vacanza Team and individual contributors (see CONTRIBUTORS file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from gettext import gettext as tr
+
+from holidays.calendars import _CustomIslamicHolidays
+from holidays.calendars.gregorian import (
+    JAN,
+    FEB,
+    MAR,
+    APR,
+    MAY,
+    JUN,
+    JUL,
+    AUG,
+    SEP,
+    OCT,
+    THU,
+    FRI,
+    SAT,
+    _timedelta,
+)
+from holidays.constants import PUBLIC, SCHOOL, WORKDAY
+from holidays.groups import InternationalHolidays, IslamicHolidays
+from holidays.observed_holiday_base import ObservedHolidayBase, FRI_TO_NEXT_WORKDAY
+
+
+class Yemen(ObservedHolidayBase, InternationalHolidays, IslamicHolidays):
+    """Yemen holidays.
+
+    References:
+        * [Presidential Decree Law No. (8) of 1996](https://web.archive.org/web/20230905142654/http://yemen-nic.info/db/laws_ye/detail.php?ID=11476)
+        * [Presidential Decree Law No. (2) of 2000](https://web.archive.org/web/20140714152707/http://www.presidentsaleh.gov.ye/showlaws.php?_lwbkno=3&_lwptno=2&_lwnmid=4)
+        * <https://en.wikipedia.org/wiki/Public_holidays_in_Yemen>
+        * <https://ar.wikipedia.org/wiki/قائمة_العطل_الرسمية_في_اليمن>
+        * <https://web.archive.org/web/20160806072136/http://marebpress.net/news_details.php?lang=arabic&sid=55120>
+        * <https://web.archive.org/web/20250505071425/https://yemenembassy.it/festivities/>
+        * <https://web.archive.org/web/20250113111659/https://www.timeanddate.com/holidays/yemen/2025?hol=9>
+        * <https://web.archive.org/web/20250103115023/http://www.yemenpost.net/Detail123456789.aspx?ID=3&SubID=7132&MainCat=3>
+    """
+
+    country = "YE"
+    default_language = "ar"
+    # %s (estimated).
+    estimated_label = tr("%s (مُقدَّر)")
+    # %s (observed).
+    observed_label = tr("(ملاحظة) %s")
+    # %s (observed, estimated).
+    observed_estimated_label = tr("(مُقدَّر ملاحظة) %s")
+    # The Republic of Yemen was declared on 22 May 1990.
+    start_year = 1991
+    supported_categories = (PUBLIC, SCHOOL, WORKDAY)
+    supported_languages = ("ar", "en_US")
+
+    def __init__(self, islamic_show_estimated: bool = True, *args, **kwargs):
+        """
+        Args:
+            islamic_show_estimated:
+                Whether to add "estimated" label to Islamic holidays name
+                if holiday date is estimated.
+        """
+        InternationalHolidays.__init__(self)
+        IslamicHolidays.__init__(
+            self, cls=YemenIslamicHolidays, show_estimated=islamic_show_estimated
+        )
+        kwargs.setdefault("observed_rule", FRI_TO_NEXT_WORKDAY)
+        super().__init__(*args, **kwargs)
+
+    def _populate_public_holidays(self):
+        self.dts_observed = set()
+        # Yemen switches from THU-FRI to FRI-SAT on Aug 15, 2013
```

Comment:
```suggestion
        dts_observed = set()

        # Yemen switches from THU-FRI to FRI-SAT on Aug 15, 2013
```
Local variable is enough.